src/Spectre.Data.Tests/RoiDictionaryTests.cs

```diff
@@ -0,0 +1,195 @@
+            Assert.AreEqual(actual: obtainedRoi.RoiPixels[2].YCoordinate, expected: DataStub.ReadRoiDataset.RoiPixels[2].YCoordinate, message: "Coordinates doesn't match.");
+
+            Assert.IsNull(nonExistentRoi, "nonExistentRoi != null");
+        }
+
+        [Test]
+        public void Add_adds_roi_properly()
+        {
+            var roiDictionaryService = new RoiDictionary();
+            roiDictionaryService.SetDirectoryPath(DataStub.TestDirectoryPath);
+            roiDictionaryService.LoadAllRois();
+
+            roiDictionaryService.Remove("addtestfile");
+            var nonExistentRoi = roiDictionaryService.GetRoiOrDefault("addtestfile");
+
+            Assert.IsNull(nonExistentRoi, "nonExistentRoi != null");
+
+            var pathForNonExistingFile = Path.Combine(DataStub.TestDirectoryPath, "addtestfile" + ".png");
+
+            Assert.IsFalse(File.Exists(pathForNonExistingFile));
+
+            roiDictionaryService.Add(DataStub.AddRoiDataset);
+
+            var obtainedRoi = roiDictionaryService.GetRoiOrDefault("addtestfile");
+
+            Assert.AreEqual(actual: obtainedRoi.Name, expected: "addtestfile");
+            Assert.IsTrue(File.Exists(Path.Combine(DataStub.TestDirectoryPath, "addtestfile" + ".png")));
+        }
+
+        [Test]
+        public void Remove_removes_from_dictionary_properly()
+        {
+            var roiDictionaryService = new RoiDictionary();
+            roiDictionaryService.SetDirectoryPath(DataStub.TestDirectoryPath);
+            roiDictionaryService.LoadAllRois();
+
+            Assert.IsTrue(File.Exists(Path.Combine(DataStub.TestDirectoryPath, "addtestfile" + ".png")));
+
+            roiDictionaryService.Remove("addtestfile");
+
+            var obtainedRoi = roiDictionaryService.GetRoiOrDefault("addtestfile");
+
+            Assert.IsNull(obtainedRoi);
+
+            Assert.IsFalse(File.Exists(Path.Combine(DataStub.TestDirectoryPath, "addtestfile" + ".png")));
+
+            roiDictionaryService.Add(DataStub.AddRoiDataset);
+        }
+
+        [Test]
+        public void Roi_constructor_throws_when_pixels_out_of_image()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(
+                code: () =>
+                {
+                    new Roi(
+                        "randomname",
+                        10,
+                        10,
+                        new List<RoiPixel>
+                        {
+                            new RoiPixel(15, 6),
+                            new RoiPixel(1, 15)
+                        });
+                });
+        }
+
+        [Test]
+        public void LoadAllRois_loads_properly()
+        {
+            RoiDictionary roiDictionaryService = new RoiDictionary();
+            roiDictionaryService.SetDirectoryPath(DataStub.TestDirectoryPath);
+
+            roiDictionaryService.LoadAllRois();
+
+            var obtainedRoi = roiDictionaryService.GetRoiOrDefault("image1");
```

Comment:
Check [this](https://github.com/nunit/docs/wiki/Values-Attribute). It would be nice-to-have for your convenience next time. (I post this just for informational purposes - if you feel like applying it, that's fine, but it's not required for me).